server/container/guice/guice-utils/src/test/java/org/apache/james/utils/PackageNameTest.java

```
@@ -0,0 +1,72 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.Test;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+class PackageNameTest {
+    @Test
+    void shouldMatchBeanContract() {
+        EqualsVerifier.forClass(PackageName.class)
+            .verify();
+    }
+
+    @Test
+    void ofShouldThrowWhenNull() {
+        assertThatThrownBy(() -> PackageName.of(null))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void ofShouldThrowWhenEmpty() {
+        assertThatThrownBy(() -> PackageName.of(""))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void getNameShouldReturnSuppliedValue() {
+        String name = "org.apache.MyClass";
+        assertThat(PackageName.of(name).getName())
+            .isEqualTo(name);
+    }
+
+    @Test
+    void ofShouldSanitizeStartingDot() {
+        assertThat(PackageName.of(".org.apache.MyClass").getName())
+            .isEqualTo("org.apache.MyClass");
+    }
+
+    @Test
+    void ofShouldSanitizeEndingDot() {
+        assertThat(PackageName.of("org.apache.MyClass.").getName())
+            .isEqualTo("org.apache.MyClass");
+    }
+
+    @Test
+    void ofShouldSanitizeDoubleDot() {
```

why?

---

it could hide an error in the string provided by the user, no?